File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        n=len(gas)
        if sum(gas)-sum(cost)<0:
            return -1
        i=0
        tank=0
        start=0
        while True:
            if start==i-n and tank>=0:
                return i%n
            if tank<0:
                tank=0
                start=i
            tank+=gas[i%n]-cost[i%n]
            i+=1
            
            
        

File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        long sumt=0;
        int l=gas.length;
        for(int i=0; i<l; ++i){
            gas[i]-=cost[i];
            sumt+=gas[i];
        }
        if(sumt<0) return -1;
        int start=0, i=0, tank=0;
        while(true){
            if(start == i-l) return i%l;
            tank += gas[i%l];
            i++;
            if(tank<0){
                tank=0;
                start=i;
            }
        }
    }
}